src/main/java/frc/robot/subsystems/SwerveDrivetrain.java

```diff
@@ -105,9 +106,6 @@ public void periodic() {
         }else {
             poseEstimator.updatePosition(gyroValue);
         }
-        if (Constants.SWERVE_DEBUG) {
```

Comment:
lines 103-107 seem expensive, maybe put some in separate thread?

Reply:
putting in another branch

Reply:
(the gyro is in another branch) the pose estimator would be hard to put into another thread because wpilib is not know to be thread safe and calls a lot of built in methods. I'm sure its possible but hopefully the pose estimator is not taking to much time. However, I can take a brief look if there is a section of it that would be easy to thread